java/hsfs/src/main/java/com/logicalclocks/hsfs/engine/FeatureViewEngineBase.java

```diff
@@ -79,35 +79,35 @@ public static List<TrainingDatasetFeature> makeLabelFeatures(QueryBase query, Li
       for (Feature feat : (List<Feature>) query.getLeftFeatures()) {
         labelWithPrefixToFeature.put(feat.getName(), feat.getName());
         labelWithPrefixToFeatureGroup.put(feat.getName(),
-            (new FeatureGroupBaseForApi(null, feat.getFeatureGroupId())));
+            (new StreamFeatureGroup(null, feat.getFeatureGroupId())));
```

Comment:
why change it to `StreamFeatureGroup`?

Reply:
FeatureGroupBaseForApi was used as a dummy class for API calls. In hsfs only abstract classes were present and it is not initializable. Since we added now support for  StreamFeatureGroup, it can do the job